engine/bsql_engine/io/io.pyx

```diff
@@ -433,8 +441,12 @@ cpdef runSkipDataCaller(table, queryPy):
       for i in range(names.size()):
           decoded_names.append(names[i].decode('utf-8'))
 
-      df = cudf.DataFrame(CudfXxTable.from_unique_ptr(blaz_move(dereference(resultSet).cudfTable), decoded_names)._data)
-      return_object['metadata'] = df 
+      dfs = []
+      for i in range(dereference(resultSet).cudfTables.size()):
+        dfs.append(cudf.DataFrame(CudfXxTable.from_unique_ptr(blaz_move(dereference(resultSet).cudfTables[i]), decoded_names)._data))
+
+      return_object['metadata'] = dfs
```

Comment:
`runSkipDataPython`, much like `parseMetadataPython` are expected to only return one partition. We should add an assert here for that and what this function returns should be the same as it was before. Lets not make this more complicated than it needs to by making it fix the same multi-partition output as `runQueryCaller`

Reply:
I removed that code. The modifications now are only for the runQuery function.